spec/services/action_builder_spec.rb

```diff
@@ -43,4 +43,77 @@ def initialize(params)
     mab.build_action
     expect(mab.previous_action).to eq(found_action)
   end
+
+  describe 'permitted_keys' do
+
+    let(:mab) { MockActionBuilder.new(page_id: page.id, email: member.email) }
+
+    it 'returns symbols' do
+      expect(mab.permitted_keys.map(&:class).uniq).to eq [Symbol]
+    end
+
+    it 'does not include the id' do
+      expect(mab.permitted_keys).not_to include(:id)
+    end
+
+    it 'includes all the other keys of member' do
+      expect(mab.permitted_keys).to include(:email, :country, :first_name, :last_name, :city, :postal, :title, :address1, :address2, :actionkit_user_id)
```

Comment:
I really love it when a spec tells me exactly what I should expect out of a method :)